### PR TITLE
Article: Java Control Statements

### PR DESCRIPTION
**Pull request by Shaif95**
Java Control Statements article in #534 . Please review this and let me know the changes required.

### CODE REVIEW

File: Java-Control-Statements.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
+#Java Control Statements :
+
+**In** any programming languege, the flow of program is neccessary for executing the program and achieving desired results.
```
Comment (alayek) [inline Java-Control-Statements.md]: > **In** any programming languege, the flow of program is neccessary for executing the program and achieving desired results.

Note sure if this highlights the pint of having control flow in a programming language. Here's how I think I would highlight the importance of control flow

> In any programming language, the control over normal flow of program is necessary to make dynamic decisions at runtime and achieving desired results.

File: Java-Control-Statements.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+shown below :
+
+```
+if(condition)
+statement1;
+else
+statement2;
+```
+
+In the above code snippet, the `condition` is first resolved. If it turns out to be true, then the statements that are inside the if
```
Comment (alayek) [inline Java-Control-Statements.md]: > In the above code snippet, the `condition` is first ~~resolved~~ evaluated.

Comment (alayek) [inline Java-Control-Statements.md]: > If it turns out to be ~~true~~`true`

Comment (alayek) [inline Java-Control-Statements.md]: > inside the ~~if~~ `if`

File: Java-Control-Statements.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+```
+if(condition)
+statement1;
+else
+statement2;
+```
+
+In the above code snippet, the `condition` is first resolved. If it turns out to be true, then the statements that are inside the if 
+condition are executed(`statement1 in this case`). If the `condition` turns out to be false, then the statements inside else are
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- `statement1 in this case`
+ `statement1` in this case
```

Comment (alayek) [inline Java-Control-Statements.md]: > turns out to be ~~false~~ `false`

Comment (alayek) [inline Java-Control-Statements.md]: > statements inside ~~else~~ `else` are

File: Java-Control-Statements.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+```
+if(condition)
+statement1;
+else
+statement2;
+```
+
+In the above code snippet, the `condition` is first resolved. If it turns out to be true, then the statements that are inside the if 
+condition are executed(`statement1 in this case`). If the `condition` turns out to be false, then the statements inside else are 
+executed(`statement2, in this case`).
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- `statement2, in this case`
+ `statement2`, in this case
```

File: Java-Control-Statements.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+statement2;
+```
+
+In the above code snippet, the `condition` is first resolved. If it turns out to be true, then the statements that are inside the if 
+condition are executed(`statement1 in this case`). If the `condition` turns out to be false, then the statements inside else are 
+executed(`statement2, in this case`). 
+
+**Example :**
+
+Here is an example usage which states whether a number is an `even number or an odd number`. Even numbers are divisible by two
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- whether a number is an `even number or an odd number`
+ whether a number is an even number or an odd number
```

Comment (alayek) [inline Java-Control-Statements.md]: > Even numbers are divisible by ~~two~~ 2

File: Java-Control-Statements.md [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+and hence leave a remainder of zero. The remainder on dividing the number by two can be obtained using the `modulo (%) mathematical
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- by two can be obtained using the `modulo (%) mathematical
+ by 2 can be obtained using the `modulo (%)` mathematical

File: Java-Control-Statements.md [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+operator`. 
+
+```
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- ```
+ ```java
```

File: Java-Control-Statements.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+   
+if ( num%2==0){ 
+    System.out.println("Number is even");
+}
+else{
+    System.out.println("Number is odd");
+       }
+   }
+}
+```
```
Comment (alayek) [inline Java-Control-Statements.md]: Create a [REPL](https:/repl.it) code snippet and link it here. It would be easier to test out the code at one click.

@koustuvsinha maybe you can show him how to do this?

File: Java-Control-Statements.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+
+**Output : Number is odd**
```
Comment (alayek) [inline Java-Control-Statements.md]: Not the standard way to write output. Usually it's written as inline comments with code snippets.

Or you can write it as an English sentence. No need to bold.

File: Java-Control-Statements.md [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+####Nested if-else
+
+`Nested if-else` is very common in programming. Here one if statement is a target of another or subsequent `if-else statements`.
```
Comment (alayek) [inline Java-Control-Statements.md]: ```diff
- `Nested if-else`
+ Nested `if-else`
```

File: Java-Control-Statements.md [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+
+**Example :**
+
+```
```
Comment (alayek) [inline Java-Control-Statements.md]: add Java Syntax higlighting, as above.

File: Java-Control-Statements.md [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+    } else {
+        if (num == 3) {
+            System.out.println("Three");      //second if statement
+        } else {
+            System.out.println("Number is greater than three");
+        }
+    }
+}
+```
+
```
Comment (alayek) [inline Java-Control-Statements.md]: Same as [above](https://github.com/FreeCodeCamp/wiki/pull/792/files#r60974983)

File: Java-Control-Statements.md [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+**Output : Number is greater than three**
```
Comment (alayek) [inline Java-Control-Statements.md]: Same as [above](#r60975113)

File: Java-Control-Statements.md [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+} else if (num == 2) {
+    System.out.println("Two");
+} else if (num == 3) {
+    System.out.println("Three");
+} else {
+    System.out.println("Numbers greater than three cannot be processed by this code");
+}
+```
+
+**Output : Three**
```
Comment (alayek) [inline Java-Control-Statements.md]: Same as [above](#r60975113)

File: Java-Control-Statements.md [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+if (num == 1) {
+    System.out.println("One");
+} else if (num == 2) {
+    System.out.println("Two");
+} else if (num == 3) {
+    System.out.println("Three");
+} else {
+    System.out.println("Numbers greater than three cannot be processed by this code");
+}
+```
```
Comment (alayek) [inline Java-Control-Statements.md]: Same as [above](#r60974983)

File: Java-Control-Statements.md [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+         case 'F' :
+            System.out.println("Failed");
+            break;
+         default :
+            System.out.println("Invalid grade");
+      }
+      System.out.println("Your grade is " + grade);
+   }
+}
+```
```
Comment (alayek) [inline Java-Control-Statements.md]: Same as [above](#r60974983)

File: Java-Control-Statements.md [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+} else if (num == 3) {
+    System.out.println("Three");
+} else {
+    System.out.println("Numbers greater than three cannot be processed by this code");
+}
+```
+
+**Output : Three**
+
+Here the if statement is executed from `top-down`. As soon as one of the `if condition` is true the statements associated with it are
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- Here the if statement
+ Here, the `if` statement
```

Honestly, you don't need to start with _Here_.

File: Java-Control-Statements.md [hunk 18]
```diff
@@ -0,0 +1,216 @@
...
+executed, and the rest of the ladder is bypassed. If none of the conditions are true then the final `else statement` is executed as a
```
Comment (alayek) [inline Java-Control-Statements.md]:
```diff
- executed,
+ has been executed,
```

File: Java-Control-Statements.md [hunk 19]
```diff
@@ -0,0 +1,216 @@
...
+`default condition`.
+
+###switch
+
+The `switch` staement is Java's multiway branching statement. It provides an easy way to dispatch execution to a desired path based on
+the value of an expression. It is actually  a better alternative than using large `if-else statements`.
```
Comment (alayek) [inline Java-Control-Statements.md]: It's not always. If statements can compare between two variables. Switch can compare only between a variable and several possible **constant** values.

Comment (alayek) [inline Java-Control-Statements.md]: From [Java Specification](http://docs.oracle.com/javase/specs/jls/se7/html/jls-14.html#jls-14.11),

> When the switch statement is executed, first the Expression is evaluated. If the Expression evaluates to null, a `NullPointerException` is thrown and the entire switch statement completes abruptly for that reason.

Basically you would need to test if the variable you are passing to `switch` is not `null` before doing so.

File: Java-Control-Statements.md [hunk 20]
```diff
@@ -0,0 +1,216 @@
...
+
+**Example :**
+
+```
```
Comment (alayek) [inline Java-Control-Statements.md]: Syntax highlight for Java.

File: Java-Control-Statements.md [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+            break;
+         default :
+            System.out.println("Invalid grade");
+      }
+      System.out.println("Your grade is " + grade);
+   }
+}
+```
+
+**Output :**
```
Comment (alayek) [inline Java-Control-Statements.md]: Same as [above](https://github.com/FreeCodeCamp/wiki/pull/792/files#r60975113)

File: Java-Control-Statements.md [hunk 22]
```diff
@@ -0,0 +1,216 @@
...
+Your grade Excellent!
+Your grade Good
+Your grade Passed
+Your grade Failed
+```
+
+
+###Summary:
+
+The `switch` differs from the `if` in that `switch` can only test for equality, whereas if can evaluate any type of Boolean expressio
```
Comment (alayek) [inline Java-Control-Statements.md]: > Boolean ~~expressio~~ expression

File: Java-Control-Statements.md [hunk 23]
```diff
@@ -0,0 +1,216 @@
...
+n. Also, note that two case constants can't have identical values in same `switch`.
+A `switch statement` is usually more efficient than a set of `nested ifs`. In case of `if-else` the Java-compiler would have a check,
```
Comment (alayek) [inline Java-Control-Statements.md]: This is debatable. So present at least two sides of the argument. What do you mean by **efficient**? Runtime performance in terms of speed or memory consumed?

**Comment by alayek** [comment]
@koustuvsinha please take a look.

@abhisekp @theflametrooper might wanna check for typo, Wiki style rules (camperbot friendly etc..).

**Comment by alayek** [comment]
@Shaif95 great article! Thanks for contributing to FCC Wiki.

However, this is your second time contributing, so we expect some improvements in the quality of your submissions. Following would be my suggestions:
- Create a new branch in your clone. Don't use master.
- Read the whole article before submitting. Basically proofreading.
- No need to contribute such long big articles, if you think it's too much to tackle at once. You can break these down in 3-4 articles.
- Know what to put inside code highlight block, and what not to.

It's ok. It won't happen in a day. But if you keep practicing, the maturity would show. If you need any help, contact us [here](https://gitter.im/FreeCodeCamp/Wiki).

File: Java-Control-Statements.md [hunk 24]
```diff
@@ -0,0 +1,216 @@
...
+Java has two selection statements : `if-else` and `switch`.
+
+###if-else
+
+`if-else` is a conditional branch statement that helps us to choose between two available choices. The general syntax of `if-else` is
+shown below :
+
+```
+if(condition)
+statement1;
```
Comment (abhisekp) [inline Java-Control-Statements.md]: a 2 or 4 space indentation would be nice : )

**Comment by Rafase282** [comment]
### This article was part of Java Basics and it has been split so this will need to me merged into one file soon.

File: Java-Control-Statements.md [hunk 25]
```diff
@@ -0,0 +1,223 @@
+#Java Control Statements :
+
+In any programming language, the control over normal flow of program is necessary to make dynamic decisions at runtime and achieving desired results. `Control statements` manage the flow of execution of the program and decide the branching of logical flow. Java's program control statements can be put into following categories: `selection` , `iteration` and `jump statement` . `Selection statement` allows a program to choose different path of executions depending on an expression. `Iteration` allows a program to repeat one or more statements. `Jump statement` is used to transfer control to another part of the program. As `iteration` and `jump statements` are already covered in `Java-loops`, here we would discuss `Selection statements`.
```
Comment (abhisekp) [inline Java-Control-Statements.md]: I think it will be appropriate to use backticks for codes and for emphasising general words like *Control statements*, *jump statements*, etc, _italics_ will do.

File: Java-Control-Statements.md [hunk 26]
```diff
@@ -0,0 +1,223 @@
...
+
+##Java's selection statement :
+
+`Selection statements` allow a program to choose different executions based on the outcome of an `expression` or `state of a variable
+`.
+Java has two selection statements : `if-else` and `switch`.
+
+###if-else
+
```
Comment (abhisekp) [inline Java-Control-Statements.md]: Usually people write `if...else` statement instead of `if-else`. (*just nitpicking, no big deal*)

**Comment by Rafase282** [comment]
We already have an article about [Java Control Flow](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Java-Control-Flow). Also there is https://github.com/FreeCodeCamp/wiki/pull/798 I would suggest you get in contact with the other person and work on improving the existing article.

Thank you.
